java/runtime/src/main/java/org/ray/runtime/RayMultiWorkerNativeRuntime.java

```diff
@@ -0,0 +1,175 @@
+package org.ray.runtime;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import org.ray.api.RayActor;
+import org.ray.api.RayObject;
+import org.ray.api.RayPyActor;
+import org.ray.api.WaitResult;
+import org.ray.api.function.RayFunc;
+import org.ray.api.id.ObjectId;
+import org.ray.api.id.UniqueId;
+import org.ray.api.options.ActorCreationOptions;
+import org.ray.api.options.CallOptions;
+import org.ray.api.runtime.RayRuntime;
+import org.ray.api.runtimecontext.RuntimeContext;
+import org.ray.runtime.config.RayConfig;
+import org.ray.runtime.config.RunMode;
+import org.ray.runtime.generated.Common.WorkerType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RayMultiWorkerNativeRuntime implements RayRuntime {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RayMultiWorkerNativeRuntime.class);
+
+  private final int workerCount;
+  private final Thread[] threads;
+  private final RayNativeRuntime[] runtimes;
+  private final ThreadLocal<RayNativeRuntime> currentThreadRuntime = new ThreadLocal<>();
+
+  private CountDownLatch shutdownCountDownLatch = new CountDownLatch(1);
+
+  public RayMultiWorkerNativeRuntime(RayConfig rayConfig) {
+    Preconditions.checkState(
+        rayConfig.runMode == RunMode.CLUSTER && rayConfig.workerMode == WorkerType.WORKER);
+    Preconditions.checkState(rayConfig.numWorkersPerProcess > 0,
+        "numWorkersPerProcess must be greater than 0.");
+    workerCount = rayConfig.numWorkersPerProcess;
+    runtimes = new RayNativeRuntime[workerCount];
+    threads = new Thread[workerCount];
+
+    LOGGER.info("Starting {} workers.", workerCount);
+
+    for (int i = 0; i < workerCount; i++) {
+      int finalI = i;
+      threads[i] = new Thread(() -> {
+        RayNativeRuntime runtime = new RayNativeRuntime(rayConfig);
+        runtimes[finalI] = runtime;
+        currentThreadRuntime.set(runtime);
+        runtime.run();
+      });
+    }
+  }
+
+  public void run() {
+    for (int i = 0; i < workerCount; i++) {
+      threads[i].start();
+    }
+    for (int i = 0; i < workerCount; i++) {
+      try {
+        threads[i].join();
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Override
+  public void shutdown() {
+    shutdownCountDownLatch.countDown();
+    for (int i = 0; i < workerCount; i++) {
+      try {
+        runtimes[i].shutdown();
+        threads[i].join();
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  private RayNativeRuntime getCurrentRuntime() {
+    RayNativeRuntime currentRuntime = currentThreadRuntime.get();
+    Preconditions.checkNotNull(currentRuntime,
+        "RayRuntime is not set on current thread. If the task runs with multiple threads, please ensure Ray.asyncClosure is called when starting new threads.");
```

Comment:
1) What about ``"RayRuntime is not set on current thread. If you want to use Ray API in your own threads, please wrap your `Runnable`s or `Callable`s with `Ray.asyncClosure`. ``
2) This line is too long, break it into multiple lines (use `+`).

Reply:
Done.